security/plugins/secgroup.py

```diff
@@ -13,17 +13,61 @@
 #    License for the specific language governing permissions and limitations
 #    under the License.
 
+import logging
+
 from security import base
 
+from keystoneauth1 import identity
+from keystoneauth1 import session
+from neutronclient.v2_0 import client
+
+UNSAFE_RULES = [
+    {
+        "protocol": ("0", "tcp", "udp"),
+        "remote_ip_prefix": (None, "0.0.0.0/0"),
+        "port_range_min": None,
+        "port_range_max": None,
+    },
+]
+
+
+def match_rule(pattern, rule):
+    for field, value in pattern.items():
+        if isinstance(value, tuple):
+            if rule[field] not in value:
+                return None
+        else:
+            if rule[field] != value:
+                return None
```

Comment:
nit:`False` feels a bit more coherent.

Side note: I've spent a couple of minutes going through all the ifs, to see what the function actually does. However I don't have a good alternative for it =)